src/be/app.py

```diff
@@ -24,20 +29,52 @@ def index():
 # serve the frontend's static assets
 @app.get("/static/{path:path}")
 def static(path: str):
+    print(f"static path: {path}")
     return fa.responses.FileResponse(
         path=_fe_dir / path,
     )
 
+# I don't know why this needs to be here but it breaks without it :)
+@app.get("/turn_on")
+def hello():
+    bulb.turn_on()
+    return "Hello from the server!"
 
 @app.get("/api/v1/status")
 def status():
+    #Use the `bulb` variable to get the properties of the LED strip, such as its name, power status, brightness, and color. Return these properties as an instance of the `Status` class.
+    bulb_name_variable = bulb.get_properties()['name']
+    power_status = bulb.get_properties()['power']
+    brightness = bulb.get_properties()['bright']
+    colour = bulb.get_properties()['rgb']
+    if bulb_name_variable is None:
+        bulb_name_variable = 'Unknown'
+    print(f"Bulb name: {bulb_name_variable}, Power status: {power_status}, Brightness: {brightness}, Colour: {colour}")
     return Status(
-        bulb_name="TODO",
+        bulb_name= bulb_name_variable,
+        power_status='on',
+        brightness=0,
+        colour=(0xFF, 0xFF, 0xFF)
     )
 
 @app.post("/api/v1/turn_on")
 def turn_on():
-    # TODO turn bulb on here
-    return "Hello from the server! implement this function"
+    bulb.turn_on() 
+    return "Turned on light"
+
+@app.post("/api/v1/turn_off")
+def turn_off():
+    bulb.turn_off()
+    return "Turned off light"
+
+bulb = None
+
+def discover_bulb():
+    bulbs = discover_bulbs()
+    bulb_ip = bulbs[0]['ip']
```

Comment:
To avoid accidentally turning other peoples yeelights on and off here you could get a light by it's name (by just getting the first bulb in the list that matches on the name property or something)

You'll have to get your light bulb and give it a name though, you could do that in a jupyter notebook or debug session (or however you like) with the bulb.set_name() function

Reply:
May need to do another quick session on debugs / jupyter - Cannot for the life of me get it to change the bulb name